data-prepper-plugin-framework/src/test/java/org/opensearch/dataprepper/plugin/PluginBeanFactoryProviderTest.java

```diff
@@ -112,8 +113,32 @@ void testCreatePluginSpecificContext() {
     void testCreatePluginSpecificContext_with_empty_array() {
         when(context.getParent()).thenReturn(context);
         final PluginBeanFactoryProvider objectUnderTest = createObjectUnderTest();
-        BeanFactory beanFactory = objectUnderTest.createPluginSpecificContext(new Class[]{});
+        BeanFactory beanFactory = objectUnderTest.createPluginSpecificContext(new Class[]{}, null);
         assertThat(beanFactory, notNullValue());
         assertThrows(NoSuchBeanDefinitionException.class, ()->beanFactory.getBean(TestComponent.class));
     }
+
+    @Test
+    void testCreatePluginSpecificContext_with_empty_array_with_plugin_config() {
+        when(context.getParent()).thenReturn(context);
+        final PluginBeanFactoryProvider objectUnderTest = createObjectUnderTest();
+        TestPluginConfiguration config = new TestPluginConfiguration();
+        BeanFactory beanFactory = objectUnderTest.createPluginSpecificContext(new Class[]{}, config);
+        assertThat(beanFactory, notNullValue());
+        assertThrows(NoSuchBeanDefinitionException.class, ()->beanFactory.getBean(TestComponent.class));
```

Comment:
I think these assertions are somewhat problematic because they are testing that two somewhat arbitrary beans do not exist. Is there another API we can use from either `BeanFactory` or `GenericApplicationContext`? For example, is there a list beans, or count beans method? We know these should be empty.

Reply:
I couldn't find any other better way. BeanFactory doesn't have methods to provide registered bean count.

Reply:
Thank you for the detailed explanation of this possible approach 👍 I modified the test to assert in this way. please review.